storage_tiering.cpp

```diff
@@ -817,7 +817,7 @@ namespace irods {
                 _source_resource,
                 low_tier_resource_name,
                 get_verification_for_resc(comm_, low_tier_resource_name),
-                false,
+                get_preserve_replicas_for_resc(comm_, low_tier_resource_name),
```

Comment:
> Actually... should this be `_source_resource` instead of `low_tier_resource_name`?
> 
> We are intending to preserve the replica on the resource _from which_ we are restaging (i.e. source resource), so it seems that we should check the configuration on that resource rather than the minimum restage tier (i.e. destination resource).

Hi,
You might be right.
But the fix breaks the configuration. :(
If the parameter is not set it will now give an error and do nothing at all. No restaging of objects to an higher tier.
But it will get the object directly from the lower tier.

Will implement a better fix.

Reply:
@cookie33 - I'm guessing that you got this working?

Reply:
Yup. This works as advertised in the code.
And that is why I have test: `test_preserve_replicas_works_with_restage_when_replica_only_exists_in_last_tier__issue_233`

Reply:
Cool, thanks!